mypy/build.py

```diff
@@ -1521,8 +1596,10 @@ def __init__(self,
                 # difference and just assume 'builtins' everywhere,
                 # which simplifies code.
                 file_id = '__builtin__'
-            path = find_module(file_id, manager.lib_path)
+            path = find_module(file_id, manager.lib_path, manager.options.python)
             if path:
+                if any((path.startswith(d) for d in package_dirs_cache)):
```

Comment:
Couldn't this break if the package_dirs_cache contains /a/b and the package is in a/bc?

Reply:
Good point, I will fix this.

Reply:
I'm actually less concerned about this. Since the directory will always be of the form eg `/path/to/site-packages`, any string concatenated to that will either be wrong (`/path/to/site-packagesfoo`?), or unsafe (as in it's unsafe to add site-packages to your path).

Reply:
`os.path.commonpath([d, path]) == d` would be correct, as would `os.path.relpath(path, d).startswith('..')`